improver/calibration/ensemble_calibration.py

```diff
@@ -562,114 +557,201 @@ def __repr__(self):
+        """
+        spatial_coords_and_dims = []
+        for axis in ["x", "y"]:
+            historic_coord_points = historic_forecast.coord(axis=axis).points
+            coord_point = np.median(historic_coord_points)
+            coord_bounds = [historic_coord_points[0], historic_coord_points[-1]]
+            new_coord = historic_forecast.coord(axis=axis).copy(
+                points=coord_point, bounds=coord_bounds
+            )
+            spatial_coords_and_dims.append((new_coord, None))
+        return spatial_coords_and_dims
+
+    def _combine_data_and_metadata(
+        self, optimised_coeffs, historic_forecast, aux_coords_and_dims, attributes
+    ):
+        """Create a cubelist by combining the optimised coefficients and the
+        appropriate metadata. The units of the alpha and gamma coefficients
+        match the units of the historic forecast. If the predictor is the
+        realizations, then the beta coefficient cube contains a realization
+        coordinate.
+
+        Args:
+            optimised_coeffs (numpy.ndarray)
+            historic_forecast (iris.cube.Cube)
+            aux_coords_and_dims (list of tuples)
+                List of tuples of the format [(coord, dim), (coord, dim)]
+            attributes (dict)
+
+        Returns:
+            cubelist (iris.cube.CubeList):
+                CubeList constructed using the coefficients provided and using
+                metadata from the historic_forecast cube. Each cube within the
+                cubelist is for a separate EMOS coefficient e.g. alpha, beta,
+                gamma, delta.
+        """
+        cubelist = iris.cube.CubeList([])
+        for optimised_coeff, coeff_name in zip(optimised_coeffs, self.coeff_names):
+            coeff_units = "1"
+            if coeff_name in ["alpha", "gamma"]:
+                coeff_units = historic_forecast.units
+            dim_coords_and_dims = []
+            if self.predictor.lower() == "realizations" and coeff_name == "beta":
+                dim_coords_and_dims = [
+                    (historic_forecast.coord("realization").copy(), 0)
+                ]
+            cube = iris.cube.Cube(
+                optimised_coeff,
+                long_name=f"emos_coefficient_{coeff_name}",
+                units=coeff_units,
+                dim_coords_and_dims=dim_coords_and_dims,
+                aux_coords_and_dims=aux_coords_and_dims,
+                attributes=attributes,
+            )
+            cubelist.append(cube)
+        return cubelist
+
+    def create_coefficients_cubelist(self, optimised_coeffs, historic_forecast):
+        """Create a cubelist for storing the coefficients computed using EMOS.
 
         .. See the documentation for examples of these cubes.
         .. include:: extended_documentation/calibration/
            ensemble_calibration/create_coefficients_cube.rst
 
         Args:
-            optimised_coeffs (list):
+            optimised_coeffs (numpy.ndarray):
```

Comment:
Looks like this might have been changed by mistake?

Reply:
I've updated this.